webpack/assets/javascripts/react_app/components/BookmarkForm/BookmarkForm.js

```diff
@@ -1,76 +1,204 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import * as Yup from 'yup';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import {
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  Checkbox,
+  ActionGroup,
+  HelperText,
+  HelperTextItem,
+  FormHelperText,
+} from '@patternfly/react-core';
 import { useDispatch } from 'react-redux';
-import { noop } from '../../common/helpers';
-import ForemanForm from '../common/forms/ForemanForm';
-import TextField from '../common/forms/TextField';
-import { translate as __ } from '../../common/I18n';
-import { maxLengthMsg, requiredMsg } from '../common/forms/validators';
-import { submitForm } from '../../redux/actions/common/forms';
+import { sprintf, translate as __ } from '../../common/I18n';
+import { APIActions } from '../../redux/API';
+import './bookmarkForm.css';
 
 const BookmarkForm = ({
   url,
   controller,
-  onCancel,
   searchQuery,
   setModalClosed,
   bookmarks,
 }) => {
   const dispatch = useDispatch();
-  const existsNamesRegex = new RegExp(
-    `^(?!(${bookmarks.map(({ name }) => name).join('|')})$).+`
+  const existingNames = bookmarks.map(({ name }) => name);
+
+  const NAME_MAX_LENGTH = 255;
+  const NAME_MIN_LENGTH = 1;
+  const SUCCESS_TYPE = 'success';
+  const ERROR_TYPE = 'error';
+
+  const nameTooLongMsg = sprintf(
+    __('name is too long, max %s'),
+    NAME_MAX_LENGTH
   );
-  const bookmarkFormSchema = Yup.object().shape({
-    name: Yup.string()
-      .max(...maxLengthMsg(254))
-      .required(requiredMsg())
-      .matches(existsNamesRegex, {
-        excludeEmptyString: true,
-        message: __('name already exists'),
-      }),
-    query: Yup.string()
-      .max(...maxLengthMsg(4096))
-      .required(requiredMsg()),
-  });
+  const nameExists = __('name already exists');
+  const nameShort = __("can't be blank");
+  const blankInputErrMsg = __("Can't be blank");
```

Comment:
Should be the same, and sentence case
<img width="288" height="245" alt="image" src="https://github.com/user-attachments/assets/63dc2916-0011-490c-9846-6fc8843852aa" />